src/main/scala/org/allenai/ike/IkeBatchAPI.scala

```diff
@@ -0,0 +1,226 @@
+
+/** Created by matt on 8/1/16.
+  *
+  * This lets you run some of the pattern matching within other Scala code or from the command line
+  *
+  * You can run it as: sbt "run-main org.allenai.ike.IkeCommandlineRunner"
+  *
+  * - OR -
+  *
+  * val ikeCmd = new IkeBatchAPI()
+  * val extracts = ikeCmd.extract(extractionConfig) // extractionConfig = your config file
+  *
+  */
+class IkeBatchAPI extends Logging {
+  val usageLogger = LoggerFactory.getLogger("Usage")
+
+  logger.debug("Starting IkeToolActor")
+  // this is just here to force logger initialization
+
+  val config = ConfigFactory.load
+
+  // Create instances of embedding based similar phrase searchers
+  val word2vecPhrasesSearcher =
+    new EmbeddingBasedPhrasesSearcher(ConfigFactory.load()[Config]("word2vecPhrasesSearcher"))
+  val tableExpander = new SimilarPhrasesBasedTableExpander(word2vecPhrasesSearcher)
+
+  val searchApps = config.getConfigList("IkeToolWebapp.indices").asScala.map { config =>
+    config.getString("name") -> SearchApp(config)
+  }.toMap
+
+  /** This is the entry point for this API (though I left everything else public as well, just in case someone
+    * wanted to use them for whatever purposes). This method takes a configuration file which defines the patterns,
+    * as well as a configuration option to say which index holds hte actual content to run your patterns against.
+    *
+    * NOTE!!! We assume that there is only 1 index that you are using and care about!
+    *
+    * TODO: This should probably be refactored so you pull the search index from the regular IKE config file,
+    * rather than managing two config files, and then you might be able to associate patterns with an index
+    * rather than making this assumption
+    *
+    * The config file with the patterns looks like:
+    *
+    * -------------------------
+    * ExtractionPatterns = {
+    *   patterns = [
+    *     {
+    *       name = "result-percent",
+    *       pattern = "CD {%|percent|per cent|pct}"
+    *     },
+    *     {
+    *       name = "treatements",
+    *       pattern = "{were given|treated with|received|receiving} {CD|NN|JJ|IN}+"
+    *     }
+    *   ]
+    * }
+    *
+    * IndexWithContent = "RCTs"
+    * -------------------------
+    *
+    * @param extractionConfigFile The config file
+    * @return A List of ApiExtractionResult case class objects
+    */
+  def extract(extractionConfigFile: String): List[ApiExtractionResult] = {
+
+    val searchApps = config.getConfigList("IkeToolWebapp.indices").asScala.map { config =>
+      config.getString("name") -> SearchApp(config)
+    }.toMap
+
+    val indexWithContent = ConfigFactory.load(extractionConfigFile).getString("IndexWithContent")
+
+    /*
+     need to make a rule for each config
+     also, load up rules from a DB. Or always load up from DB??
+     */
+    val extractionPatterns = loadPatternsFromConfig(extractionConfigFile) // ++ loadDictionaryPatterns(extractionConfigFile)
```

Comment:
Taking `Seq[NamedPattern]` as input seems most appropriate here given the following definition: `case class NamedPattern(name: String, pattern: String)`